src/main/java/practice/CandidateValidator.java

```diff
@@ -1,5 +1,43 @@
 package practice;
 
-public class CandidateValidator {
-    //write your code here
+import java.util.function.Predicate;
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> {
+    private final String slashSeparator = "-";
+
+    @Override
+    public boolean test(Candidate candidate) {
+        if (validCandidateAge(candidate) && validNationality(candidate)
+                && validPeriodInUkr(candidate) && candidate.isAllowedToVote()) {
+            return true;
+        }
+        return false;
+    }
+
+    public static void main(String[] args) {
+        Candidate fred = new Candidate(39, "German", true, "2000-2019");
+        fred.setName("Fred");
+        CandidateValidator candidateValidator = new CandidateValidator();
+        System.out.println(candidateValidator.test(fred));
+    }
+
+    private boolean validCandidateAge(Candidate candidate) {
+        return candidate.getAge() >= 35;
+    }
+
+    private boolean validNationality(Candidate candidate) {
+        return candidate.getNationality().equals("Ukrainian");
+    }
+
+    private boolean validPeriodInUkr(Candidate candidate) {
+        int lastTimeInUkr = Integer.parseInt(candidate.getPeriodsInUkr()
+                .split(slashSeparator)[1]);
+        int startYearInUkr = Integer.parseInt(candidate.getPeriodsInUkr()
+                .split(slashSeparator)[0]);
+        if (lastTimeInUkr - startYearInUkr > 10) {
+            return true;
+        }
+        return false;
+    }
```

Comment:
Don't complicate if-else clause
```suggestion
            return lastTimeInUkr - startYearInUkr > 10);
    }
```